tools/meson-vcs-tag.sh

```diff
@@ -15,4 +15,16 @@ fi
 # Apparently git describe has a bug where it always considers the work-tree
 # dirty when invoked with --git-dir (even though 'git status' is happy). Work
 # around this issue by cd-ing to the source directory.
-cd "$dir" && git describe --abbrev=7 --dirty=+ 2>/dev/null | sed 's/^v//' || echo "$fallback"
+cd "$dir"
+
+version=""
+isgit="`git rev-parse --git-dir`"
+if [ "x$isgit" = "x.git" ]; then
+    version="`git describe --abbrev=7 --dirty=+ 2>/dev/null | sed 's/^v//'`"
+fi
+
+if [ "x$version" = "x" ]; then
+    version="$fallback"
+fi
+
+echo "$version"
```

Comment:
So... the think with "x$var" = "x" is ugly. This is a work-around for some very old shells, and we don't do this in our scripts. Please just use `[ -z "$version" ]` or `[ -n "$version" ]` as appropriate to check if the string is empty, and a normal comparison otherwise.

So what about the following:
```sh
cd "$dir"
isgit="$(git rev-parse --git-dir)"
[ "$isgit" != ".git" ] && git describe --abbrev=7 --dirty=+ 2>/dev/null | sed 's/^v//' || echo "$fallback"
```
?

Reply:
Avoid build systemd release tarball (without .git directory) and where there is parent .git directory,
this situation will get wrong git version info.

A little cosmetic (!= -> ==):
```shell
cd "$dir"
isgit="$(git rev-parse --git-dir)"
[ "$isgit" == ".git" ] && git describe --abbrev=7 --dirty=+ 2>/dev/null | sed 's/^v//' || echo "$fallback"
```